### ISSUE TITLE
KeyError: 'LD_LIBRARY_PATH'. And libbitsandbytes_cpu.so is ELF

### ISSUE THREAD

**Issue by lewischi** [TimDettmers/bitsandbytes#576]
Just started experiencing this error.
1. Like someone suggested, I could export LD_LIBRARY_PATH in .zshrc, but I'm not sure what to set it to.
2. Also not sure why "libbitsandbytes_cpu.so" is ELF instead of mach-o

Macbook Pro:
Intel silicon
Intel UHD Graphics 630 1536 MB

```
➜ poetry show bitsandbytes
 name         : bitsandbytes
 version      : 0.39.0
 description  : k-bit optimizers and matrix multiplication routines.
```

```
➜ pwd
/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes
➜ file libbitsandbytes_cpu.so
libbitsandbytes_cpu.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=c696c46cacfdec3e6746fc8d399147ba829c8b9c, not stripped
```
```
===================================BUG REPORT===================================
Welcome to bitsandbytes. For bug reports, please run

python -m bitsandbytes

 and submit this information together with your error trace to: https://github.com/TimDettmers/bitsandbytes/issues
================================================================================
bin /Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so
/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/cextension.py:34: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU quantization are unavailable.
  warn("The installed version of bitsandbytes was compiled without GPU support. "
'NoneType' object has no attribute 'cadam32bit_grad_fp32'
CUDA SETUP: Loading binary /Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so...
dlopen(/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so, 0x0006): tried: '/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so' (not a mach-o file), '/System/Volumes/Preboot/Cryptexes/OS/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so' (no such file), '/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so' (not a mach-o file)
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
++++++++++++++++++ BUG REPORT INFORMATION ++++++++++++++++++
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++

++++++++++++++++++ /usr/local CUDA PATHS +++++++++++++++++++
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda116_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda112_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda115_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda111_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda110.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda114.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda120.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda115.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda121.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda118_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda121_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda111.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda116.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda112.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda120_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda113.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda117.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda110_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda114_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda118.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda113_nocublaslt.so
/usr/local/lib/python3.11/site-packages/bitsandbytes/libbitsandbytes_cuda117_nocublaslt.so

+++++++++++++++ WORKING DIRECTORY CUDA PATHS +++++++++++++++


++++++++++++++++++ LD_LIBRARY CUDA PATHS +++++++++++++++++++
Traceback (most recent call last):
  File "/Users/lewis.chi/.pyenv/versions/3.10.6/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/lewis.chi/.pyenv/versions/3.10.6/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/__main__.py", line 95, in <module>
    generate_bug_report_information()
  File "/Users/lewis.chi/Library/Caches/pypoetry/virtualenvs/ml-triton-server-pRkgzG-4-py3.10/lib/python3.10/site-packages/bitsandbytes/__main__.py", line 66, in generate_bug_report_information
    lib_path = os.environ['LD_LIBRARY_PATH'].strip()
  File "/Users/lewis.chi/.pyenv/versions/3.10.6/lib/python3.10/os.py", line 679, in __getitem__
    raise KeyError(key) from None
KeyError: 'LD_LIBRARY_PATH'
```

**Comment by amanbirs** [comment]
Same issue. I used version 0.39.1 and that seemed to work without this bug.

**Comment by TimDettmers** [comment]
What OS are you running? It seems you have no CUDA installed. This is a requirement to use bitsandbytes.

**Comment by TimDettmers** [comment]
This has been fixed in f4996978db44102bc30b2e97820ab63d108c234c. This will be pushed soon to pip. Thank you for reporting.
